hub/core/version_control/test_version_control.py

```diff
@@ -0,0 +1,304 @@
+    ds.create_tensor("abc")
+    ds.create_tensor("img")
+    for i in range(10):
+        ds.img.append(i * np.ones((100, 100, 3)))
+    a = ds.commit("first")
+
+    ds.img[7] *= 2
+
+    assert (ds.img[6].numpy() == 6 * np.ones((100, 100, 3))).all()
+    assert (ds.img[7].numpy() == 2 * 7 * np.ones((100, 100, 3))).all()
+    assert (ds.img[8].numpy() == 8 * np.ones((100, 100, 3))).all()
+    assert (ds.img[9].numpy() == 9 * np.ones((100, 100, 3))).all()
+
+    assert (ds.img[2].numpy() == 2 * np.ones((100, 100, 3))).all()
+
+    b = ds.commit("second")
+
+    # going back to first commit
+    ds.checkout(a)
+
+    assert (ds.img[7].numpy() == 7 * np.ones((100, 100, 3))).all()
+
+    ds.checkout("another", create=True)
+
+    ds.img[7] *= 3
+
+    # and not 6 * 7 as it would have been, had we checked out from b
+    assert (ds.img[7].numpy() == 3 * 7 * np.ones((100, 100, 3))).all()
+
+    ds.commit("first2")
+
+    ds.checkout("main")
+    assert (ds.img[7].numpy() == 2 * 7 * np.ones((100, 100, 3))).all()
+    ds.log()
+
+
+@enabled_datasets
+def test_auto_checkout_bug(ds):
+    ds.create_tensor("abc")
+    ds.abc.extend([1, 2, 3, 4, 5])
+    a = ds.commit("it is 1")
+    ds.abc[0] = 2
+    b = ds.commit("it is 2")
+    c = ds.checkout(a)
+    d = ds.checkout("other", True)
+    ds.abc[0] = 3
+    e = ds.commit("it is 3")
+    ds.checkout(b)
+    ds.abc[0] = 4
+    f = ds.commit("it is 4")
+    g = ds.checkout(a)
+    ds.abc[0] = 5
+    dsv = ds[0:3]
+    dsv.abc[0] = 5
+    h = ds.commit("it is 5")
+    i = ds.checkout(e)
+    ds.abc[0] = 6
+    tsv = ds.abc[0:5]
+    tsv[0] = 6
+    j = ds.commit("it is 6")
+    ds.log()
+    ds.checkout(a)
+    assert dsv.abc[0].numpy() == 1
+    assert ds.abc[0].numpy() == 1
+    ds.checkout(b)
+    assert ds.abc[0].numpy() == 2
+    ds.checkout(c)
+    assert ds.abc[0].numpy() == 1
+    ds.checkout(d)
+    assert ds.abc[0].numpy() == 3
+    ds.checkout(e)
+    assert ds.abc[0].numpy() == 3
+    ds.checkout(f)
+    assert ds.abc[0].numpy() == 4
+    ds.checkout(g)
+    assert ds.abc[0].numpy() == 1
+    ds.checkout(h)
+    assert ds.abc[0].numpy() == 5
+    ds.checkout(i)
+    assert ds.abc[0].numpy() == 3
+    ds.checkout(j)
+    assert ds.abc[0].numpy() == 6
+    ds.checkout("main")
+    assert ds.abc[0].numpy() == 2
+    ds.abc[0] = 7
+    ds.checkout("copy", True)
+    assert ds.abc[0].numpy() == 7
+    ds.checkout("other")
+    assert ds.abc[0].numpy() == 3
+
+
+# @enabled_datasets
+# def test_read_mode(ds):
+#     ds.create_tensor("abc")
+#     ds.checkout("second", create=True)
+#     with pytest.raises(ReadOnlyModeError):
+#         ds.commit("first")
+#     with pytest.raises(ReadOnlyModeError):
+#         ds.checkout("third", create=True)
+#     with pytest.raises(ReadOnlyModeError):
+#         ds.abc.append(10)
+
+
+@enabled_datasets
+def test_checkout_address_not_found(ds):
+    with pytest.raises(CheckoutError):
+        ds.checkout("second")
+
+
+@enabled_datasets
```

Comment:
i don't think we need to use enabled_datasets for VC, memory and/or local should suffice right?